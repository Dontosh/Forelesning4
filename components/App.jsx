import React, { useState } from "react";

export function ListMovies() {
  return (
    <div>
      <h1>Movie Database</h1>
      <ul>
        <li>
          <h1>Movie 1</h1>
        </li>
        <li>
          <h1>Movie 2</h1>
        </li>
      </ul>
    </div>
  );
}

export function NewMovieForm({ onAddMovie }) {
  const [title, setTitle] = useState("");
  const [year, setYear] = useState("");
  const [plot] = useState("");

  function handleSubmit() {
    onAddMovie({ title, year, plot });
  }

  return (
    <form onSubmit={handleSubmit}>
      <h1>Add new movie</h1>
      <div>
        Title:{" "}
        <input
          data-testid="title"
          value={title}
          onChange={(e) => setTitle(e.target.value)}
        />
      </div>
      <div>
        Year:{" "}
        <input
          data-testid="year"
          value={year}
          onChange={(e) => setYear(e.target.value)}
        />
      </div>
      <div>
        Plot: <textarea data-testid="plot" />
      </div>
    </form>
  );
}