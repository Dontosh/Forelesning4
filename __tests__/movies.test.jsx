import { render } from "react-dom";
import { ListMovies } from "../components/App";
import { NewMovieForm } from "../components/App";
import React from "react";
import { Simulate } from "react-dom/test-utils";

describe("movies application", () => {
  it("shows movie list", () => {
    const element = document.createElement("div");
    render(<ListMovies />, element);
    expect(element.querySelector("h1").innerHTML).toEqual("Movie Database");
    expect(element.innerHTML).toMatchSnapshot();
  });

  it("show new movie form", () => {
    const element = document.createElement("div");
    render(<NewMovieForm />, element);
    expect(element.innerHTML).toMatchSnapshot();
  });

  it("submit new movie", () => {
    const onAddMovie = jest.fn();

    const element = document.createElement("div");
    render(<NewMovieForm onAddMovie={onAddMovie} />, element);

    Simulate.change(element.querySelector("[data-testid=title]"), {
      target: { value: "Movie 1" },
    });
    Simulate.change(element.querySelector("[data-testid=year]"), {
      target: { value: "2022" },
    });
    Simulate.submit(element.querySelector("form"));
    expect(onAddMovie).toHaveBeenCalledWith({
      title: "Movie 1",
      year: "2022",
      plot: "",
    });
  });
});
