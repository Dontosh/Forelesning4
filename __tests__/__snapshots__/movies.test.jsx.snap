// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`movies application show new movie form 1`] = `"<form><h1>Add new movie</h1><div>Title: <input data-testid=\\"title\\" value=\\"\\"></div><div>Year: <input data-testid=\\"year\\" value=\\"\\"></div><div>Plot: <textarea data-testid=\\"plot\\"></textarea></div></form>"`;

exports[`movies application shows movie list 1`] = `"<div><h1>Movie Database</h1><ul><li><h1>Movie 1</h1></li><li><h1>Movie 2</h1></li></ul></div>"`;
